### DISCUSSION TITLE
Git log command fails in action

### DISCUSSION THREAD

**Discussion by Nuckerr** [community/community#55079] [Actions] [answered]
### Select Topic Area

Question

### Body

I have this git command being run in my gradle task:
```kotlin
executable("git")
args("log",  "$startHash..$endHash", "--format=format:%h %s")
```
where the range ends up being: `b94771f86d0a1027b9fbbda02879f30b76aab2c2..HEAD`
Now, when I run this locally, it runs fine, however in the github action it gives this error:
```
fatal: Invalid revision range b94771f86d0a1027b9fbbda02879f30b76aab2c2..HEAD
(...)
> Failed to calculate the value of property 'changelog'.
   > Process 'command 'git'' finished with non-zero exit value 128
```
Perhaps the checkout is only pulling the latest git information? Not sure, could really do with some help

**Comment by airtower-luna** [comment, marked as answer]
Is the start commit (`b94771f86d0a1027b9fbbda02879f30b76aab2c2`) actually available during the workflow run? By default `actions/checkout` retrieves only the latest commit, not the history.

Assuming it's that, the best way to fix it depends on context, the easiest one is probably to tell `actions/checkout` to fetch the whole history (`fetch-depth: 0`). The simple method will likely waste time & bandwidth though (the larger the repository the more), so it might be better to fetch only the part you actually need.

**Comment by Nuckerr** [comment]
well that commit is stored in a properties file, if theres any way of extracting that into the action then sure, but i dont think that will be easy. However, the action has `workflow_dispatch` so I might be able to add an input or something

**Comment by airtower-luna** [comment]
If you know the date of the commit, it's relatively simple using `git fetch` with [`--shallow-since`](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---shallow-sinceltdategt).

Using [`git fetch --shallow-exclude=BRANCH_OR_TAG`](https://git-scm.com/docs/git-fetch#Documentation/git-fetch.txt---shallow-excludeltrevisiongt) with a branch or tag pointing at a commit *before* the one you need should work, too.

Otherwise it might get more complex, there doesn't seem to be an option to fetch everything since a specific commit. Whether it's worth the complexity really depends on how big your repository is, if it's small `fetch-depth: 0` might be just fine.
